New Resource: `azurerm_mobile_network_data_network`; New DataSource: `azurerm_mobile_network_data_network` (#20338)
hashicorp/terraform-provider-azurerm: website/docs/r/mobile_network_data_network.html.markdown

  name     = "example-resources"
  location = "East Us"
}

resource "azurerm_mobile_network" "example" {
  name                = "example-mn"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  mobile_country_code = "001"
  mobile_network_code = "01"
}

resource "azurerm_mobile_network_data_network" "example" {
  name              = "example-mndn"
  mobile_network_id = azurerm_mobile_network.example.id
  location          = azurerm_resource_group.example.location
  description       = "example description"

  tags = {
    key = "value"
  }

}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) Specifies the name which should be used for this Mobile Network Data Network. Changing this forces a new Mobile Network Data Network to be created.

* `mobile_network_id` - (Required) Specifies the ID of the Mobile Network. Changing this forces a new Mobile Network Data Network to be created.

* `location` - (Required) Specifies the Azure Region where the Mobile Network Data Network should exist. Changing this forces a new Mobile Network Data Network to be created.

> * `description` - (Optional) An optional description for this data network.

Reviewer: ```suggestion
* `description` - (Optional) A description of this Mobile Network Data Network.
```

* `tags` - (Optional) A mapping of tags which should be assigned to the Mobile Network Data Network.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported:

* `id` - The ID of the Mobile Network Data Network.

> 
> 

Reviewer: this isn't a big deal, but the linter will send a PR to fix this otherwise, so we should sort this whilst we're making other changes:

```suggestion
```
